### ISSUE TITLE
installer err - cannot locate program input point DiscardVirtualMemory on dynamic link library KERNEL32.dll

### ISSUE THREAD

**Issue by jzy-chitong56** [SMUnlimited/AMAI#392]
Unable to locate program input point DiscardVirtualMemory on dynamic link library KERNEL32.dll

I'm not sure what the official English prompt is, so I'll post it in Chinese for you to proofread the machine translation above
`无法定位程序输入点 DiscardVirtualMemory 于动态链接库 KERNEL32.dll 上`

This issue seems to be common in WIN7, affecting all electronic packaging

![cb3510a8450e553327fd1df7570e18e7](https://github.com/user-attachments/assets/98ef90e7-87b0-436c-becd-c768c4514785)

**Comment by SMUnlimited** [comment]
Based on translation installer still compiling for win10 and above somehow. Should be fixed as per https://github.com/SMUnlimited/AMAI/issues/317

Unless trying to use 32bit instead of a 64 bit system?

**Comment by paulo101977** [comment]
> Unable to locate program input point DiscardVirtualMemory on dynamic link library KERNEL32.dll
> 
> I'm not sure what the official English prompt is, so I'll post it in Chinese for you to proofread the machine translation above `无法定位程序输入点 DiscardVirtualMemory 于动态链接库 KERNEL32.dll 上`
> 
> This issue seems to be common in WIN7, affecting all electronic packaging
> 
> @paulo101977 Excuse me, but I don't know how to handle it. Sorry
> 
> ![cb3510a8450e553327fd1df7570e18e7](https://private-user-images.githubusercontent.com/48715223/373156747-98ef90e7-87b0-436c-becd-c768c4514785.jpeg?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3Mjc5ODMwMjEsIm5iZiI6MTcyNzk4MjcyMSwicGF0aCI6Ii80ODcxNTIyMy8zNzMxNTY3NDctOThlZjkwZTctODdiMC00MzZjLWJlY2QtYzc2OGM0NTE0Nzg1LmpwZWc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQxMDAzJTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MTAwM1QxOTEyMDFaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT1iOTE2ZmY5ZmE0ODY4NTY2M2U1NWVmMzkxYmNhZWYwZjk5OGQ5ZGQ3ZWMwYzcxYjcxZDQ3MzkyMjNkYTVjODk1JlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCJ9.Zsl7VDEqWqqrueH3G0E2Btjbjm29M4ndLjiYh4X0_kM)

Apologies for the delay. I didn't understand the context, but is it not due to the system version?

**Comment by paulo101977** [comment]
> Based on translation installer still compiling for win10 and above somehow. Should be fixed as per #317
> 
> Unless trying to use 32bit instead of a 64 bit system?

I thought the same. Maybe because the system is 32 bits. I think we would have to generate a 32-bit artifact and test it.

**Comment by jzy-chitong56** [comment]
According to my personality    •᷄ࡇ•᷅  
Then don't test it, just package it as 64 bit . Write in the description that only 64 bit is supported  o(￣ヘ￣o＃)
 
This kind of dependency package is too troublesome to check

**Comment by paulo101977** [comment]
> According to my personality •᷄ࡇ•᷅ Then don't test it, just package it as 64 bit . Write in the description that only 64 bit is supported o(￣ヘ￣o＃)
> 
> This kind of dependency package is too troublesome to check

There is a way to compile using 32 bits, but this has to be done in deploy: npm run electron:build -- --ia32
So, we would have to run the build both to generate the 64-bit artifact and another for 32 bits.

**Comment by jzy-chitong56** [comment]
> 
> There is a way to compile using 32 bits, but this has to be done in deploy: npm run electron:build -- --ia32
> 
> So, we would have to run the build both to generate the 64-bit artifact and another for 32 bits.


If you're saying you don't need to worry about the configuration file and just modify the packaging command, I've already packed the package, but I don't know where to find a 32-bit system for testing. Sorry to ask for help


I like to package attachments on the workflow because I don't know how to install environments on a computer
https://github.com/jzy-chitong56/Electron/actions/runs/11191617667

**Comment by jzy-chitong56** [comment]
it was found that the program is still 64 bit, so compilation cannot generate a 32-bit program

**Comment by jzy-chitong56** [comment]
> We could create two build processes. One for 32bits and one for 64bits. However, I had never tested this process for 32 bits, I don't know if it depends on some artifact, lib, etc.

no need 
My R&D team used the same tools to package our company's own software, but it still cannot run on WIN7 X64
So it should still be a compatibility issue with WIN7

**Comment by paulo101977** [comment]
> > We could create two build processes. One for 32bits and one for 64bits. However, I had never tested this process for 32 bits, I don't know if it depends on some artifact, lib, etc.
> 
> no need My R&D team used the same tools to package our company's own software, but it still cannot run on WIN7 X64 So it should still be a compatibility issue with WIN7

Yes, that could be it too. And perhaps even Electron's lack of support for Windows 7.

**Comment by jzy-chitong56** [comment]
https://blog.csdn.net/u012864465/article/details/135888718

here said , @SMUnlimited 
electron use 21.4.4
electron-builder use 21.2.0

**Comment by SMUnlimited** [comment]
Someone needs to try various tests before we downgrade. 

Try electron downgrade only and see if that works on Win7
Downgrade electron-builder only and see if that works on Win7 as there is not hints this is incompatible with Win7.
Try both and see if that works.

Official documentation states that electron 22 is the last version to work with Win7.
The result of tests will be the right fix.

**Comment by jzy-chitong56** [comment]
you can try my branches
Some branches can be automatically packaged as attachments, just modify the code and I will download it locally for testing later, because modifying this is not my expertise, but I suspect I cannot find 32-bit
